fetch/interface.go

```diff
@@ -13,32 +13,32 @@ type requester interface {
 	Request(context.Context, p2p.Peer, []byte, func([]byte), func(error)) error
 }
 
-type malfeasanceHandler interface {
+type MalfeasanceValidator interface {
 	HandleSyncedMalfeasanceProof(context.Context, p2p.Peer, []byte) error
 }
 
-type atxHandler interface {
+type AtxValidator interface {
 	HandleAtxData(context.Context, p2p.Peer, []byte) error
 }
 
-type blockHandler interface {
+type BlockValidator interface {
 	HandleSyncedBlock(context.Context, p2p.Peer, []byte) error
 }
 
-type ballotHandler interface {
+type BallotValidator interface {
 	HandleSyncedBallot(context.Context, p2p.Peer, []byte) error
 }
 
-type proposalHandler interface {
+type ProposalValidator interface {
 	HandleSyncedProposal(context.Context, p2p.Peer, []byte) error
 }
 
-type txHandler interface {
+type TxValidator interface {
 	HandleBlockTransaction(context.Context, p2p.Peer, []byte) error
 	HandleProposalTransaction(context.Context, p2p.Peer, []byte) error
 }
 
-type poetHandler interface {
+type PoetValidator interface {
 	ValidateAndStoreMsg(context.Context, p2p.Peer, []byte) error
 }
```

Comment:
Do you know what is the reason that these all are not just a single interface like:
```go
type P2PHandler interface {
    Handle(context.Context, p2p.Peer, []byte) error
}
```
?

Reply:
if you take a look at node.go, these interface are provided by different instances in different packages with different validation logic. i cannot think of a quick way to resolve this in a clean way.

Reply:
Rename the methods or wrap them :)

Reply:
renaming methods is not an option without big change, as the same handler can handle multiple data (see TxValidator)
wrapping them is possible, but that basically is a layer of indirection.

i.e. you will have an 
```go
type dataValidator interface {
   validateA()
   validateB()
   validateC()
}
```
then fills the following struct with the gossip handlers for A, B, and C
```go
type validatorImpls struct {
   ValidatorAImpl
   ValidatorBImpl
   ValidatorCImpl
}
```
and pass this validatorImpls to fetcher. it's not clear to me that this approach is better. let me know if you think differently and i will follow up with another PR